NebulaModel/DataStructures/PlayerData.cs

```diff
@@ -45,6 +48,20 @@ public void Serialize(INetDataWriter writer)
             Rotation.Serialize(writer);
             BodyRotation.Serialize(writer);
             Mecha.Serialize(writer);
+            writer.Put(Appearance != null);
```

Comment:
I think you could avoid this line if you use the length to tell whether appearance is there or not

Reply:
I just took the logic from here: https://github.com/hubastard/nebula/blob/3c7da1bd62a9e58d77eac2f898c07f2fc2d9ccec/NebulaModel/DataStructures/MechaData.cs#L48